### PR TITLE
Implemented a portable mode to lmms

### PR DESCRIPTION
**Pull request by SeleDreams**
I've implemented this portable mode based on this issue https://github.com/LMMS/lmms/issues/5201

this is my first contribution to the project I tried to respect the programming style while refactoring a bit.

the overall modifications are : 
- I have added an environment variable PORTABLE_MODE, when LMMS is built with this environment variable, it will default its directory to the directory where it is executed (in portable mode, the lmms directory has been renamed to lmms-workspace in order to avoid name conflict on linux between the executable and the directory)

- I have fixed the typo from the function availabeVstEmbedMethods to availableVstEmbedMethods

- I have refactored the constructor of ConfigManager to be more readable

- I have added the CLion files autogenerated when opening a cmake project to the gitignore

### CODE REVIEW

**Comment by PhysSong** [comment]
First of all, thank you for your contribution! It seems like your editor is using 4 spaces for indentation, but it should be a tab according to our coding convention.

I also suggest adding a runtime check because people don't need to rebuild LMMS to make existing builds portable. For example, OBS checks for `portable_mode.txt` and runs in portable mode if it exists.

**Comment by SeleDreams** [comment]
> First of all, thank you for your contribution! It seems like your editor is using 4 spaces for indentation, but it should be a tab according to our coding convention.
> 
> I also suggest adding a runtime check because people don't need to rebuild LMMS to make existing builds portable. For example, OBS checks for `portable_mode.txt` and runs in portable mode if it exists.

I just got CLion as my editor and as I didn't know it enough I didn't know the type of identation it used, I'll have to see if I can make it switch to tabs instead

about runtime check, that's something I thought about and I asked it on the discord when I started working on it but I didn't get responses so I went with my first idea, it shouldn't be that hard to implement though, just checking if the file exists instead of checking for the define

**Comment by SeleDreams** [comment]
Screenshot from 2020-07-07 04-22-40
found the setting, it should make future contributions easier

**Comment by SeleDreams** [comment]
I made the changes

**Comment by PhysSong** [comment]
This time, you(or the IDE) changed formatting too much. This is not about conventions, it's about irrelevant changes. Maybe it's better to reset to clean `master` and redo your works.

**Comment by SeleDreams** [comment]
as it's a bit late in this case I would probably more do it tomorrow (it's 5 AM in france)

**Comment by SeleDreams** [comment]
I just re-did it and it now only affects the changes I made

File: src/core/ConfigManager.cpp [hunk 1]
```diff
@@ -51,68 +51,31 @@ static inline QString ensureTrailingSlash(const QString & s )
...
+	{
+		initInstalledWorkingDir();
 	}
+	m_dataDir = "data:/";
+	m_vstDir = m_workingDir + "vst/";
+	m_sf2Dir = m_workingDir + SF2_PATH;
+	m_gigDir = m_workingDir + GIG_PATH;
+	m_themeDir = defaultThemeDir();
+	if (! qgetenv("LMMS_DATA_DIR").isEmpty())
+		QDir::addSearchPath("data", QString::fromLocal8Bit(qgetenv("LMMS_DATA_DIR")));
```
Comment (PhysSong) [inline src/core/ConfigManager.cpp:70]: According to the [coding convention](https://github.com/LMMS/lmms/wiki/Coding-Conventions#braces), you should use braces here.

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,7 @@
 /build
 /target
+/.idea
+/cmake-build-debug
```
Comment (PhysSong) [inline .gitignore:4]: Please see https://github.com/LMMS/lmms/issues/5062#issuecomment-656440516, a discussion about `.gitignore`.

File: src/core/ConfigManager.cpp (hunk 1 above)
Comment (SeleDreams) [reply src/core/ConfigManager.cpp:70]: I usually always do, but as it was already like this I kept it as it was, was I supposed to change it ?

Comment (PhysSong) [reply src/core/ConfigManager.cpp:70]: I didn't know that. I'll leave it up to you.

**Comment by SeleDreams** [comment]
It should be good now

**Comment by russiankumar** [comment]
Is this PR complete?

**Comment by SeleDreams** [comment]
Yes

**Pull request merged**